recipes/cad1/task1/baseline/evaluate.py

```diff
@@ -0,0 +1,227 @@
+
+    def write_header(self):
+        """Write the header row to the CSV file."""
+        with open(self.file_name, "w", encoding="utf-8", newline="") as csv_file:
+            csv_writer = csv.writer(
+                csv_file, delimiter=",", quotechar='"', quoting=csv.QUOTE_MINIMAL
+            )
+            csv_writer.writerow(
+                [
+                    "song",
+                    "listener",
+                    "score",
+                    "l_bass",
+                    "r_bass",
+                    "l_drums",
+                    "r_drums",
+                    "l_other",
+                    "r_other",
+                    "l_vocals",
+                    "r_vocals",
+                ]
+            )
+
+    def add_result(
+        self,
+        listener: str,
+        song: str,
+        score: float,
+        instruments_scores: Dict[str, float],
+    ):
+        """Add a result to the CSV file.
+
+        Args:
+            listener (str): The name of the listener who submitted the result.
+            song (str): The name of the song that the result is for.
+            score (float): The combined score for the result.
+            instruments_scores (dict): A dictionary of scores for each instrument channel in the result.
+        """
+        logger.info(f"The combined score is {score}")
+
+        with open(self.file_name, "a", encoding="utf-8", newline="") as csv_file:
+            csv_writer = csv.writer(
+                csv_file, delimiter=",", quotechar='"', quoting=csv.QUOTE_MINIMAL
+            )
+            csv_writer.writerow(
+                [
+                    song,
+                    listener,
+                    str(score),
+                    str(instruments_scores["l_bass"]),
+                    str(instruments_scores["r_bass"]),
+                    str(instruments_scores["l_drums"]),
+                    str(instruments_scores["r_drums"]),
+                    str(instruments_scores["l_other"]),
+                    str(instruments_scores["r_other"]),
+                    str(instruments_scores["l_vocals"]),
+                    str(instruments_scores["r_vocals"]),
+                ]
+            )
+
+
+def compute_haaqi(
+    enh_signal: np.ndarray,
+    ref_signal: np.ndarray,
+    audiogram: np.ndarray,
+    audiogram_frequencies: np.ndarray,
+    fs_signal: int,
+) -> float:
+    """Compute HAAQI metric"""
+
+    haaqi_audiogram_freq = [250, 500, 1000, 2000, 4000, 6000]
+    audiogram_adjusted = np.array(
+        [
+            audiogram[i]
+            for i in range(len(audiogram_frequencies))
+            if audiogram_frequencies[i] in haaqi_audiogram_freq
+        ]
+    )
+    score, _, _, _ = haaqi_v1(
+        reference=ref_signal,
+        reference_freq=fs_signal,
+        processed=enh_signal,
+        processed_freq=fs_signal,
+        hearing_loss=audiogram_adjusted,
+        equalisation=1,
+    )
+    return score
+
+
+def set_song_seed(song: str) -> None:
```

Comment:
Good to have the argument docstring but I think this returns an `int`
```suggestion
def set_song_seed(song: str) -> int:
```